docs/flatcar-linux/kvm-libvirt.md

```diff
@@ -0,0 +1,344 @@
+
+  providers = {
+    local    = "local.default"
+    null     = "null.default"
+    template = "template.default"
+    tls      = "tls.default"
+    libvirt  = "libvirt.default"
+  }
+
+  os_image_unpacked = "file:///home/CHANGEME/flatcar_production_qemu_image.img"
+
+  ssh_keys = [
+    "ssh-rsa AAAAB3Nza...",
+  ]
+
+  asset_dir = "/home/CHANGEME/clusters/asset"
+
+  machine_domain = "vmcluster.k8s"
+  cluster_name = "vmcluster"
+  node_ip_pool = "192.168.192.0/24"
+
+  controller_count = 1
+
+}
+
+module "worker-pool-one" {
+  source = "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes/workers"
+
+  providers = {
+    local    = "local.default"
+    template = "template.default"
+    tls      = "tls.default"
+    libvirt  = "libvirt.default"
+  }
+
+  ssh_keys = "${module.controller.ssh_keys}"
+
+  machine_domain = "${module.controller.machine_domain}"
+  cluster_name = "${module.controller.cluster_name}"
+  libvirtpool = "${module.controller.libvirtpool}"
+  libvirtbaseid = "${module.controller.libvirtbaseid}"
+
+  pool_name = "one"
+
+  count = 1
+
+  kubeconfig = "${module.controller.kubeconfig}"
+
+  labels = "node.supernova.io/role=backend,node-role.kubernetes.io/backend="
+}
+```
+
+Now download the Flatcar image. The example below uses the "Edge" Flatcar channel.
+You need to increase its size depending on your use case because the initial size
+is only enough for some small pods to be loaded. Starting pods with normal-sized
+container images may already fail because the disk space will be filled.
+That's why the example below directly increases the image size by 5 GB to make
+sure that a couple of container images fit on the node.
+
+```sh
+$ sudo dnf install wget bzip2 qemu-img  # or sudo apt install wget bzip2 qemu-utils
+$ wget https://edge.release.flatcar-linux.net/amd64-usr/2135.99.0/flatcar_production_qemu_image.img.bz2
+$ bunzip2 flatcar_production_qemu_image.img.bz2
+$ qemu-img resize flatcar_production_qemu_image.img +5G
+```
+
+Now check the contents of `mycluster.tf`.
+Change at least the `ssh_keys` entry to match the content of your `~/.ssh/id_KEY.pub` and
+change the `os_image_unpacked` path to point to the image location from above.
+
+Reference the [variables docs](#variables) or the
+[variables.tf](https://github.com/kinvolk/lokomotive-kubernetes/blob/master/kvm-libvirt/flatcar-linux/kubernetes/variables.tf)
+source.
+
+## ssh-agent
+
+Initial bootstrapping requires `bootkube.service` to be started on one controller node.
+Terraform uses `ssh-agent` to automate this step.
+Add your SSH private key to `ssh-agent` so that it's unlocked if secured with a passphrase
+(you can skip this if the key way created with the GNOME Passwords and Keys tool).
+
+```sh
+ssh-add ~/.ssh/id_KEY
+ssh-add -L
+```
+
+## Apply
+
+Initialize the config directory since this is the first use with Terraform.
+
+```sh
+terraform init
```

Comment:
```
 terraform init
Initializing modules...
- module.controller
  Getting source "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes"
Error downloading modules: Error loading modules: subdir "kvm-libvirt/flatcar-linux/kubernetes" not found
```
Pending when this PR gets merged, I think this readme should still reference the branch the source is coming from to make it easy for others trying to use this.
```
source = "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes?ref=kai/terraform-libvirt"
```
Same with this:
```
terraform init
Initializing modules...
- module.controller
  Getting source "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes?ref=kai/terraform-libvirt"
- module.worker-pool-one
  Getting source "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes/workers"
Error downloading modules: Error loading modules: subdir "kvm-libvirt/flatcar-linux/kubernetes/workers" not found
```
When it is eventually approved, it can be changed to reference the right value.

Reply:
Good idea to use the branch name, thanks!